fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+                raise FunctionError(e)
+
+            # Importing gRPC proto buff files from the given interface directory.
+            try:
+                module_files = self._get_interface_file_names(_grpc_interface_file_dir, _grpc_package_name)
+                grpc_interface_module_list = self._get_grpc_interface_module(module_files, _grpc_interface_file_dir,
+                                                                             _grpc_package_name)
+                logging.debug(grpc_interface_module_list)
+
+                grpc_stub_tuple = self._get_grpc_class(grpc_interface_module_list, 'stub')
+                grpc_request_tuple = self._get_grpc_class(grpc_interface_module_list, _grpc_request_method_name.lower())
+            except Exception as e:
+                yield StatusMessage("failed to load the gRPC proto buff files : {}".format(e))
+                raise FunctionError(e)
+
+            # Converting resilient function input data into json object
+            try:
+                grpc_function_data = re.sub("'", '"', grpc_function_data)
+                grpc_function_json_data = json.loads(grpc_function_data)
+            except Exception as e:
+                raise FunctionError("Input Data must be in json formatted..!")
+
+            # Checking Communication type
+            if _grpc_communication_type.lower().strip() == "unary":
+                try:
+                    # Connecting to gRPC Server with given Authentication type
+                    stub_class_obj, grpc_channel_obj = self._gRPC_Connect_Server(_grpc_secure_connection, grpc_channel, \
+                                                                                 grpc_stub_tuple[1],
+                                                                                 _grpc_certificate_path)
+                    stub_method = self._get_method_from_stub_object(stub_class_obj, _grpc_communication_type,
+                                                                    stub_name=_grpc_rpc_stub_method_name)
+                    if grpc_function_json_data:
+                        if stub_method[1] is not None:
+                            if sys.version[0] == '2':
+                                # convert unicode strings to utf-8 for python 2.7
+                                import ast
+                                grpc_function_json_data_tmp = ast.literal_eval(json.dumps(grpc_function_json_data))
+                            else:
+                                grpc_function_json_data_tmp = grpc_function_json_data
+
+                            response_received_tmp = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data_tmp))
+
+                            # Closing the gRPC Server Connection
+                            grpc_channel_obj.close()
+                        else:
+                            raise FunctionError("Stub method not found, please specify the valid stub method.")
```

Comment:
Where does the user specify this `stub message`? We should mention this in the error

Reply:
Stub method will be generated from grpc tool based on .proto file definitions.

Reply:
i think this method would evaluate presence of stub method in gerenate pb2 files.